LayoutTests/http/tests/links/redirect-301-clear-memory-cache.html

```diff
@@ -0,0 +1,16 @@
+<html>
+<head>
+<title>301 Redirect</title>
+<script>
+  function runTest() {
+    if (window.testRunner) {
+        testRunner.waitUntilDone();
+    }
+    window.setTimeout(function() {
+      window.location = 'resources/redirect-helper-multi-redirect-clear-memory-cache.pl?301#key=foo';
```

Comment:
From what I see the flow is:
- go to resources/redirect-helper-multi-redirect-clear-memory-cache.pl?301#key=foo
- this redirects to redirect-target-multi-redirect-clear-memory-cache.html#key=foo
- this navigates to redirect-helper-multi-redirect-final.pl?301#key=bar
- this redirects to redirect-target-multi-redirect-final.html#key=bar.

The 301 redirections are not the same so cache will not be hit.
I would expect this test to pass with and without the patch given this. Can you check this?

One approach that might be simpler to write and understand could be:
- Create an iframe that loads resources/redirect-helper.py?301#key=bar
- Once loaded, verify the iframe.contentWindow.location has '#key=bar'
- Optionally clear memory cache at this point (2 tests).
- Create another iframe that loads resources/redirect-helper.py?301#key=foo
- Once loaded, verify the iframe.contentWindow.location has '#key=foo'

It would also be good to add another test where the location URL contains #key=zip, in which case both frames should have the same iframe.contentWindow.location.

Reply:
Reverified the test cases without the patch and they do pass. Going to convert to the Iframe approach you recommended.

Reply:
Fixed. Using iFrames now.


I have 2 iFrames. The first loads with the fragment. Then onload the fragment checks the iFrame hash and then goes and loads the second iFrame. I then check the second hash is correct.